samtranslator/plugins/application/serverless_app_plugin.py

```diff
@@ -84,11 +88,14 @@ def __init__(
 
     @property
     def _sar_client(self) -> BaseClient:
-        # Lazy initialization of the client- create it when it is needed
+        # Lazy initialization of the client-create it when it is needed
         if not self.__sar_client:
-            # a SAR call could take a while to finish, leaving the read_timeout default (60s).
-            client_config = Config(connect_timeout=BOTO3_CONNECT_TIMEOUT)
-            self.__sar_client = boto3.client("serverlessrepo", config=client_config)
+            if self._sar_client_creator:
+                self.__sar_client = self._sar_client_creator()
```

Comment:
```suggestion
                if self.__sar_client is None:
                    self.__sar_client = self._sar_client_creator()
```

Or do we want to always override the input `sar_client`?

Reply:
Ah never mind saw L92.